runtime/ops/io.rs

```diff
@@ -459,12 +479,47 @@ impl StdFileResource {
     f: F,
   ) -> Result<R, AnyError>
   where
-    F: FnMut(&mut StdFile) -> Result<R, AnyError>,
+    F: FnOnce(&mut StdFile) -> Result<R, AnyError>,
   {
     let resource = state.resource_table.get::<StdFileResource>(rid)?;
     resource.inner.with_file(f)
   }
 
+  pub fn with_file_and_metadata<F, R>(
+    state: &mut OpState,
+    rid: ResourceId,
+    f: F,
+  ) -> Result<R, AnyError>
+  where
+    F: FnOnce(&mut StdFile, &mut FileMetadata) -> Result<R, AnyError>,
+  {
+    let resource = state.resource_table.get::<StdFileResource>(rid)?;
+    let mut meta_data = resource.metadata.borrow_mut();
+    resource.inner.with_file(|file| f(file, &mut meta_data))
+  }
+
+  pub async fn with_file_async<F, R: Send + 'static>(
```

Comment:
```suggestion
  pub async fn with_file_blocking<F, R: Send + 'static>(
```

Seems like a better name.